packages/sit-onyx/src/components/OnyxDataGrid/features/sorting/sorting.ts

```diff
@@ -98,10 +109,17 @@ export const useSorting = <TEntry extends DataGridEntry>(options?: SortOptions<T
 
     return {
       name: SORTING_FEATURE,
-      watch: [sortState, intlCompare],
+      watch: [sortState, i18n.locale],
       mutation: {
         func: sortData,
       },
+      modifyColumns: [
+        {
+          // save final column config
+          func: (_finalConfig) => (finalConfig = _finalConfig),
+          order: Infinity,
```

Comment:
The higher the order, the earlier its executed. Don't we want to save the config as late as possible so all column modifications are already applied? So negative infinity in this case